### ISSUE TITLE
horizontalalignment 'left' and 'right' do not handle spacing consistently

### ISSUE THREAD

**Issue by jason-sachs** [matplotlib/matplotlib#4380]
horizontalalignment respects leading spaces with left-alignment, ignores trailing spaces 

```
import matplotlib.pyplot as plt

fig = plt.figure(figsize=(8,6))
ax = fig.gca()
ax.plot([0,0],[-1,1],':')
for k in xrange(5):
    spc = " "*k
    ax.text(0.0,0.1*k,spc+"%d leading spaces"%k, ha='left')
    ax.text(0.0,0.1*k,("%d trailing spaces"%k)+spc, ha='right')
```

![image](https://cloud.githubusercontent.com/assets/10506541/7327663/cc57cb52-ea83-11e4-9864-bd837f5bd411.png)

**Comment by jason-sachs** [comment]
It shouldn't ignore any spaces... otherwise why would I put them there?

**Comment by OceanWolf** [comment]
Actually I see it working fine on master...

**Comment by jason-sachs** [comment]
oh? hmm, let me grab version info...

**Comment by jason-sachs** [comment]
darnit, I'm still using 1.3.1. Let me try upgrading...

**Comment by OceanWolf** [comment]
Actually I see it working fine on master...
![figure_1](https://cloud.githubusercontent.com/assets/3758703/7327706/8e9ad2c2-eacf-11e4-9ec0-1d69989dc2aa.png)

**Comment by jason-sachs** [comment]
just tried it in 1.4.3 (latest Anaconda version), got same result. Any way to see if it has been fixed recently or if it is a platform-specific bug?

**Comment by OceanWolf** [comment]
I have just run your code on 1.4.2 and I do see the bug, so yes, it seems it has already been fixed.  Kinda new here so I don't know all the bugs that have been fixed, but yes, definitely fixed.  You can either try out the master, or wait until the next full release in a few months time.

**Comment by tacaswell** [comment]
@mdboom Do you remember fixing this?

**Comment by jason-sachs** [comment]
>You can either try out the master, or wait until the next full release in a few months time.

I'll wait; I just used the `annotate()` function with `offset points` to draw text with an offset.

It might be nice to add an `offset_points` kwarg... the `text()` function is so much easier to use than `annotate()` if you just need text and no arrows.

**Comment by petehuang** [comment]
Reconfirming the fix (this time with 1.5.3). Can we close?

**Comment by QuLogic** [comment]
Works for me too.
